cache.go

```diff
@@ -80,7 +80,12 @@ func (c *cache) Update(req *pb.FindFullHashesRequest, resp *pb.FindFullHashesRes
 		if c.pttls[fullHash] == nil {
 			c.pttls[fullHash] = make(map[ThreatDescriptor]time.Time)
 		}
-		dur := time.Duration(tm.GetCacheDuration().Seconds) * time.Second
+		var dur time.Duration
+		if tmCacheDur := tm.GetCacheDuration(); tmCacheDur != nil {
+			dur = time.Duration(tm.GetCacheDuration().Seconds) * time.Second
+		} else {
+			dur = time.Second
```

Comment:
I think zero (thus no caching) is a reasonable default in cases where the cache duration is not set. If it was an int64 field or similar it would have already defaulted to zero.